Server/src/kafka/MsgBusImpl_kafka.cpp

```diff
@@ -1165,9 +1165,9 @@ void msgBus_kafka::update_LsNode(obj_bgp_peer &peer, obj_path_attr &attr, std::l
 
             if ((uint32_t) *(node.igp_router_id+4) != 0) {
                 inet_ntop(PF_INET, node.igp_router_id+4, dr, sizeof(dr));
-                strncat(igp_router_id, "[", 1);
-                strncat(igp_router_id, dr, sizeof(dr));
-                strncat(igp_router_id, "]", 1);
+                strncat(igp_router_id, "[", sizeof(igp_router_id) - strlen(igp_router_id) - 1);
```

Comment:
this warning is more of the compiler not knowing we know better. :) I'll commit this, but will need to watch for any perf changes with regards to using strlen().

Reply:
I guess in that case the better thing to do would be to switch to `strlen`?